src/keybase/packet/alloc.iced

```diff
@@ -4,17 +4,21 @@ K = require('../../const').kb
 {Signature} = require './signature'
 {Encryption} = require './encryption'
 
+katch = (f) ->
+  try return [null, f()]
+  catch e then return [e, null]
+
 exports.alloc = ({tag, body}) ->
-  ret = err = null
-  ret = switch tag
+  [err, ret] = switch tag
     when K.packet_tags.p3skb
-      P3SKB.alloc {tag, body }
+      katch () -> P3SKB.alloc { tag, body }
     when K.packet_tags.signature
-      Signature.alloc { tag, body }
+      katch () -> Signature.alloc { tag, body }
     when K.packet_tags.encryption
-      Encryption.alloc { tag, body }
+      katch () -> Encryption.alloc { tag, body }
     else
       err = new Error "unknown packet tag: #{tag}"
+      [err, null]
       null
```

Comment:
I think this `null` needs to be removed, otherwise this compiles to
```
        default:
          err = new Error("unknown packet tag: " + tag);
          [err, null];
          return null;
```

Reply:
gah! thanks for catching that. don't know how that got in